source/portal/portal.js

```diff
@@ -24,6 +26,48 @@ var Portal = function(spec, rpcReq) {
    */
   var participants = {};
 
+  // Key is participantId, value is token ID.
+  const webTransportIds = new Map();
+  const calculateSignatureForWebTransportToken = (token) => {
+      const toSign = vsprintf('%s,%s,%s,%s,%s', [
+          token.tokenId,
+          token.transportId,
+          token.participantId,
+          token.roomId,
+          token.issueTime
```

Comment:
please include a validThrough field in the token and consider token. And provide refresh mechanism in ticketRefresh.

Reply:
It's valid throughout the session. Once the participant is disconnected from portal, the token is invalid as well.